Update git2go dependency to version 33

Hello! I'm maintaining `turbogit` on [nixpkgs](https://github.com/NixOS/nixpkgs). Currently it doesn't build because of a `libgit2` version mismatch with `git2go`. Current `git2go` version depends on `v1.1` of `libgit2`, but nixpkgs contains a `v1.3` version of `libgit2`. The version of `git2go` that supports `v1.3` of `libgit2` seems to be `v33`, as described [on git2go's readme](https://github.com/libgit2/git2go). So I would like to request a dependency update and a minor release if possible to fix this issue.

<details>
<summary>Build log</summary>
<pre><code>@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/vpl3zlpfg2i73jr651x6wnpg5i37b1y2-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
@nix { "action": "setPhase", "phase": "buildPhase" }
building
Building subPackage ./.
internal/unsafeheader
internal/cpu
runtime/internal/atomic
runtime/internal/sys
math/bits
runtime/internal/math
unicode/utf8
internal/race
sync/atomic
internal/bytealg
math
unicode
golang.org/x/sys/internal/unsafeheader
github.com/b4nst/turbogit/pkg/constants
crypto/internal/subtle
crypto/subtle
runtime
unicode/utf16
vendor/golang.org/x/crypto/cryptobyte/asn1
internal/nettrace
runtime/cgo
golang.org/x/crypto/internal/subtle
encoding
container/list
vendor/golang.org/x/crypto/internal/subtle
golang.org/x/text/encoding/internal/identifier
image/color
github.com/rivo/uniseg
internal/reflectlite
sync
internal/testlog
internal/singleflight
math/rand
errors
sort
io
internal/oserror
strconv
syscall
path
bytes
strings
golang.org/x/text/transform
reflect
golang.org/x/text/width
bufio
regexp/syntax
github.com/kballard/go-shellquote
hash
crypto
internal/syscall/unix
time
internal/syscall/execenv
crypto/internal/randutil
vendor/golang.org/x/net/dns/dnsmessage
regexp
internal/poll
io/fs
context
internal/fmtsort
encoding/binary
os
crypto/hmac
crypto/rc4
golang.org/x/crypto/blowfish
hash/crc32
vendor/golang.org/x/crypto/hkdf
golang.org/x/sys/unix
crypto/sha1
crypto/cipher
fmt
path/filepath
crypto/aes
io/ioutil
os/exec
net/url
text/template/parse
github.com/AlecAivazis/survey/v2/terminal
github.com/araddon/dateparse
github.com/mattn/go-isatty
golang.org/x/term
github.com/mattn/go-colorable
github.com/hpcloud/golor
text/template
github.com/mgutz/ansi
github.com/imdario/mergo
math/big
crypto/des
crypto/sha512
github.com/b4nst/turbogit/pkg/format
crypto/ed25519/internal/edwards25519
crypto/md5
github.com/AlecAivazis/survey/v2/core
crypto/sha256
encoding/hex
encoding/base64
encoding/pem
net
github.com/AlecAivazis/survey/v2
golang.org/x/crypto/chacha20
crypto/rand
crypto/dsa
crypto/elliptic
encoding/asn1
crypto/ed25519
crypto/rsa
vendor/golang.org/x/crypto/cryptobyte
crypto/x509/pkix
golang.org/x/crypto/curve25519/internal/field
golang.org/x/crypto/ed25519
golang.org/x/crypto/poly1305
golang.org/x/crypto/curve25519
golang.org/x/crypto/ssh/internal/bcrypt_pbkdf
log
crypto/ecdsa
github.com/whilp/git-urls
encoding/json
github.com/fatih/structs
github.com/google/go-querystring/query
github.com/pkg/errors
github.com/trivago/tgo/treflect
mime
github.com/trivago/tgo/tcontainer
mime/quotedprintable
compress/flate
vendor/golang.org/x/crypto/chacha20
vendor/golang.org/x/crypto/poly1305
vendor/golang.org/x/sys/cpu
vendor/golang.org/x/crypto/curve25519
compress/gzip
vendor/golang.org/x/text/transform
vendor/golang.org/x/crypto/chacha20poly1305
vendor/golang.org/x/text/unicode/bidi
vendor/golang.org/x/text/unicode/norm
vendor/golang.org/x/net/http2/hpack
vendor/golang.org/x/text/secure/bidirule
net/http/internal
github.com/fatih/color
flag
vendor/golang.org/x/net/idna
github.com/briandowns/spinner
golang.org/x/text/encoding
github.com/gdamore/encoding
github.com/gdamore/tcell/v2/terminfo
database/sql/driver
github.com/mattn/go-runewidth
github.com/gdamore/tcell/v2/terminfo/a/ansi
github.com/gdamore/tcell/v2/terminfo/v/vt100
github.com/gdamore/tcell/v2/terminfo/v/vt102
github.com/gdamore/tcell/v2/terminfo/v/vt220
github.com/gdamore/tcell/v2/terminfo/x/xterm
github.com/gdamore/tcell/v2/terminfo/dynamic
github.com/gdamore/tcell/v2/terminfo/a/aixterm
github.com/gdamore/tcell/v2/terminfo/base
github.com/gdamore/tcell/v2/terminfo/a/alacritty
github.com/gdamore/tcell/v2/terminfo/b/beterm
github.com/gdamore/tcell/v2/terminfo/c/cygwin
github.com/gdamore/tcell/v2/terminfo/d/dtterm
github.com/gdamore/tcell/v2/terminfo/e/emacs
github.com/gdamore/tcell/v2/terminfo/f/foot
github.com/gdamore/tcell/v2/terminfo/g/gnome
github.com/gdamore/tcell/v2/terminfo/h/hpterm
github.com/gdamore/tcell/v2/terminfo/k/konsole
github.com/gdamore/tcell/v2/terminfo/k/kterm
github.com/gdamore/tcell/v2/terminfo/l/linux
github.com/gdamore/tcell/v2/terminfo/p/pcansi
github.com/gdamore/tcell/v2/terminfo/r/rxvt
github.com/gdamore/tcell/v2/terminfo/s/screen
github.com/gdamore/tcell/v2/terminfo/s/simpleterm
github.com/gdamore/tcell/v2/terminfo/s/sun
github.com/gdamore/tcell/v2/terminfo/t/termite
github.com/gdamore/tcell/v2/terminfo/t/tmux
github.com/gdamore/tcell/v2/terminfo/v/vt320
github.com/gdamore/tcell/v2/terminfo/v/vt400
github.com/gdamore/tcell/v2/terminfo/v/vt420
github.com/gdamore/tcell/v2/terminfo/v/vt52
github.com/gdamore/tcell/v2/terminfo/w/wy50
github.com/gdamore/tcell/v2/terminfo/w/wy60
github.com/gdamore/tcell/v2/terminfo/w/wy99_ansi
github.com/gdamore/tcell/v2/terminfo/x/xfce
github.com/gdamore/tcell/v2/terminfo/x/xterm_kitty
github.com/gdamore/tcell/v2/terminfo/x/xterm_termite
github.com/lucasb-eyer/go-colorful
os/signal
github.com/gdamore/tcell/v2/terminfo/extended
github.com/ktr0731/go-fuzzyfinder/scoring
github.com/nsf/termbox-go
github.com/ktr0731/go-fuzzyfinder/matching
github.com/gdamore/tcell/v2
golang.org/x/time/rate
github.com/blang/semver/v4
encoding/csv
crypto/x509
net/textproto
vendor/golang.org/x/net/http/httpproxy
github.com/spf13/pflag
mime/multipart
vendor/golang.org/x/net/http/httpguts
github.com/ktr0731/go-fuzzyfinder
text/tabwriter
github.com/dgrijalva/jwt-go
golang.org/x/crypto/ssh
crypto/tls
github.com/spf13/cobra
github.com/libgit2/git2go/v31
net/http/httptrace
# github.com/libgit2/git2go/v31
vendor/github.com/libgit2/git2go/v31/Build_system_dynamic.go:11:3: error: #error "Invalid libgit2 version; this git2go supports libgit2 between v1.1.0 and v1.1.0"
   11 | # error "Invalid libgit2 version; this git2go supports libgit2 between v1.1.0 and v1.1.0"
      |   ^~~~~
net/http
golang.org/x/net/context/ctxhttp
github.com/hashicorp/go-cleanhttp
github.com/andygrunwald/go-jira
golang.org/x/oauth2/internal
github.com/hashicorp/go-retryablehttp
golang.org/x/oauth2
github.com/xanzy/go-gitlab</code></pre>
</details>

--- a/.github/workflows/go.yml
+++ b/.github/workflows/go.yml
@@ -24,13 +24,13 @@ jobs:
         run: |
           sudo apt update
           sudo apt install cmake
-          curl -L https://github.com/libgit2/libgit2/releases/download/v1.1.0/libgit2-1.1.0.tar.gz | tar -xz
-          mkdir -p libgit2-1.1.0/build
-          cd libgit2-1.1.0/build
+          curl -L https://codeload.github.com/libgit2/libgit2/tar.gz/refs/tags/v1.3.0 | tar -xz
+          mkdir -p libgit2-1.3.0/build
+          cd libgit2-1.3.0/build
           cmake .. -DCMAKE_INSTALL_PREFIX=/usr -DBUILD_CLAR=OFF
           sudo cmake --build . --target install
           cd -
-          rm -rf libgit2-1.1.0
+          rm -rf libgit2-1.3.0
           go get -v -t -d ./...
           if [ -f Gopkg.toml ]; then
               curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -50,15 +50,14 @@ jobs:
       - name: Get dependencies
         run: |
           sudo apt update
-          sudo apt install cmake ruby ruby-dev rubygems build-essential
-          sudo gem install --no-document fpm
-          curl -L https://github.com/libgit2/libgit2/releases/download/v1.1.0/libgit2-1.1.0.tar.gz | tar -xz
-          mkdir -p libgit2-1.1.0/build
-          cd libgit2-1.1.0/build
+          sudo apt install cmake
+          curl -L https://codeload.github.com/libgit2/libgit2/tar.gz/refs/tags/v1.3.0 | tar -xz
+          mkdir -p libgit2-1.3.0/build
+          cd libgit2-1.3.0/build
           cmake .. -DCMAKE_INSTALL_PREFIX=/usr -DBUILD_CLAR=OFF
           sudo cmake --build . --target install
           cd -
-          rm -rf libgit2-1.1.0
+          rm -rf libgit2-1.3.0
           go get -v -t -d ./...
           if [ -f Gopkg.toml ]; then
               curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
@@ -101,15 +100,14 @@ jobs:
       - name: Get dependencies
         run: |
           sudo apt update
-          sudo apt install cmake ruby ruby-dev rubygems build-essential
-          sudo gem install --no-document fpm
-          curl -L https://github.com/libgit2/libgit2/releases/download/v1.1.0/libgit2-1.1.0.tar.gz | tar -xz
-          mkdir -p libgit2-1.1.0/build
-          cd libgit2-1.1.0/build
+          sudo apt install cmake
+          curl -L https://codeload.github.com/libgit2/libgit2/tar.gz/refs/tags/v1.3.0 | tar -xz
+          mkdir -p libgit2-1.3.0/build
+          cd libgit2-1.3.0/build
           cmake .. -DCMAKE_INSTALL_PREFIX=/usr -DBUILD_CLAR=OFF
           sudo cmake --build . --target install
           cd -
-          rm -rf libgit2-1.1.0
+          rm -rf libgit2-1.3.0
           go get -v -t -d ./...
           if [ -f Gopkg.toml ]; then
               curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
@@ -152,15 +150,14 @@ jobs:
       - name: Get dependencies
         run: |
           sudo apt update
-          sudo apt install cmake ruby ruby-dev rubygems build-essential rpm
-          sudo gem install --no-document fpm
-          curl -L https://github.com/libgit2/libgit2/releases/download/v1.1.0/libgit2-1.1.0.tar.gz | tar -xz
-          mkdir -p libgit2-1.1.0/build
-          cd libgit2-1.1.0/build
+          sudo apt install cmake
+          curl -L https://codeload.github.com/libgit2/libgit2/tar.gz/refs/tags/v1.3.0 | tar -xz
+          mkdir -p libgit2-1.3.0/build
+          cd libgit2-1.3.0/build
           cmake .. -DCMAKE_INSTALL_PREFIX=/usr -DBUILD_CLAR=OFF
           sudo cmake --build . --target install
           cd -
-          rm -rf libgit2-1.1.0
+          rm -rf libgit2-1.3.0
           go get -v -t -d ./...
           if [ -f Gopkg.toml ]; then
               curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh

--- a/cmd/branch.go
+++ b/cmd/branch.go
@@ -30,7 +30,7 @@ import (
 	tugit "github.com/b4nst/turbogit/pkg/git"
 	"github.com/b4nst/turbogit/pkg/integrations"
 
-	git "github.com/libgit2/git2go/v31"
+	git "github.com/libgit2/git2go/v33"
 	"github.com/spf13/cobra"
 )
 

--- a/cmd/check.go
+++ b/cmd/check.go
@@ -29,7 +29,7 @@ import (
 
 	"github.com/b4nst/turbogit/pkg/format"
 	tugit "github.com/b4nst/turbogit/pkg/git"
-	git "github.com/libgit2/git2go/v31"
+	git "github.com/libgit2/git2go/v33"
 	"github.com/spf13/cobra"
 )
 

--- a/cmd/commit.go
+++ b/cmd/commit.go
@@ -29,7 +29,7 @@ import (
 	"github.com/AlecAivazis/survey/v2"
 	"github.com/b4nst/turbogit/pkg/format"
 	tugit "github.com/b4nst/turbogit/pkg/git"
-	git "github.com/libgit2/git2go/v31"
+	git "github.com/libgit2/git2go/v33"
 	"github.com/spf13/cobra"
 )
 

--- a/cmd/log.go
+++ b/cmd/log.go
@@ -33,7 +33,7 @@ import (
 	"github.com/b4nst/turbogit/pkg/format"
 	tugit "github.com/b4nst/turbogit/pkg/git"
 	"github.com/hpcloud/golor"
-	git "github.com/libgit2/git2go/v31"
+	git "github.com/libgit2/git2go/v33"
 	"github.com/spf13/cobra"
 )
 

--- a/cmd/tag.go
+++ b/cmd/tag.go
@@ -31,7 +31,7 @@ import (
 	"github.com/b4nst/turbogit/pkg/format"
 	tugit "github.com/b4nst/turbogit/pkg/git"
 	"github.com/blang/semver/v4"
-	git "github.com/libgit2/git2go/v31"
+	git "github.com/libgit2/git2go/v33"
 	"github.com/spf13/cobra"
 )
 

--- a/go.mod
+++ b/go.mod
@@ -11,7 +11,7 @@ require (
 	github.com/hpcloud/golor v0.0.0-20150914221010-dc1b58c471a0
 	github.com/imdario/mergo v0.3.12
 	github.com/ktr0731/go-fuzzyfinder v0.5.0
-	github.com/libgit2/git2go/v31 v31.7.0
+	github.com/libgit2/git2go/v33 v33.0.4
 	github.com/nsf/termbox-go v0.0.0-20201124104050-ed494de23a00
 	github.com/spf13/cobra v1.2.1
 	github.com/stretchr/testify v1.7.0

--- a/pkg/git/commit.go
+++ b/pkg/git/commit.go
@@ -1,6 +1,6 @@
 package git
 
-import git "github.com/libgit2/git2go/v31"
+import git "github.com/libgit2/git2go/v33"
 
 // RepoTree return the current index tree
 func RepoTree(r *git.Repository) (*git.Tree, error) {

--- a/pkg/git/remote.go
+++ b/pkg/git/remote.go
@@ -3,7 +3,7 @@ package git
 import (
 	"net/url"
 
-	git "github.com/libgit2/git2go/v31"
+	git "github.com/libgit2/git2go/v33"
 	giturls "github.com/whilp/git-urls"
 )
 

--- a/pkg/git/repo.go
+++ b/pkg/git/repo.go
@@ -3,7 +3,7 @@ package git
 import (
 	"os"
 
-	git2go "github.com/libgit2/git2go/v31"
+	git2go "github.com/libgit2/git2go/v33"
 )
 
 // Getrepo returns the repository in the current directory or an error.

--- a/pkg/git/repo_test.go
+++ b/pkg/git/repo_test.go
@@ -5,7 +5,7 @@ import (
 	"os"
 	"testing"
 
-	git "github.com/libgit2/git2go/v31"
+	git "github.com/libgit2/git2go/v33"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )

--- a/pkg/git/stage.go
+++ b/pkg/git/stage.go
@@ -3,7 +3,7 @@ package git
 import (
 	"errors"
 
-	git "github.com/libgit2/git2go/v31"
+	git "github.com/libgit2/git2go/v33"
 )
 
 // StageReady returns true if the stage is ready to be committed. Otherwise it returns false if there is nothing to commit or an error.

--- a/pkg/integrations/gitlab.go
+++ b/pkg/integrations/gitlab.go
@@ -6,7 +6,7 @@ import (
 	"strings"
 
 	tugit "github.com/b4nst/turbogit/pkg/git"
-	git "github.com/libgit2/git2go/v31"
+	git "github.com/libgit2/git2go/v33"
 	"github.com/xanzy/go-gitlab"
 )
 

--- a/pkg/integrations/jira.go
+++ b/pkg/integrations/jira.go
@@ -5,7 +5,7 @@ import (
 
 	"github.com/andygrunwald/go-jira"
 	"github.com/briandowns/spinner"
-	git "github.com/libgit2/git2go/v31"
+	git "github.com/libgit2/git2go/v33"
 )
 
 const (

--- a/pkg/integrations/jira_test.go
+++ b/pkg/integrations/jira_test.go
@@ -8,7 +8,7 @@ import (
 	"testing"
 
 	"github.com/andygrunwald/go-jira"
-	git "github.com/libgit2/git2go/v31"
+	git "github.com/libgit2/git2go/v33"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )

--- a/pkg/integrations/provider.go
+++ b/pkg/integrations/provider.go
@@ -1,7 +1,7 @@
 package integrations
 
 import (
-	git "github.com/libgit2/git2go/v31"
+	git "github.com/libgit2/git2go/v33"
 )
 
 // Provider interface abstracts cross-platform providers

--- a/pkg/test/git.go
+++ b/pkg/test/git.go
@@ -6,7 +6,7 @@ import (
 	"path/filepath"
 	"testing"
 
-	git "github.com/libgit2/git2go/v31"
+	git "github.com/libgit2/git2go/v33"
 	"github.com/stretchr/testify/require"
 )
 

